src/client/client.ts

```diff
@@ -294,59 +279,58 @@ export class GraphQLTransportWSClient {
 
     this.client.onmessage = ({ data }: {data: any}) => {
       let parsedMessage: any;
+      let reqId: number;
+
       try {
         parsedMessage = JSON.parse(data);
+        reqId = parsedMessage.id;
       } catch (e) {
         throw new Error(`Message must be JSON-parseable. Got: ${data}`);
       }
-      const reqId = parsedMessage.id;
-      if ([MessageTypes.GQL_KEEP_ALIVE, MessageTypes.GQL_INIT_SUCCESS, MessageTypes.GQL_INIT_FAIL]
-              .indexOf(parsedMessage.type) === -1 && !this.requests[reqId]) {
-        this.unsubscribe(reqId);
 
-        if (parsedMessage.type === MessageTypes.GQL_KEEP_ALIVE) {
-          return;
-        }
+      // Complete every request that are not registered on the client but are being sent by the server and are not
+      // in the allowed list
+      if (
+          [ MessageTypes.GQL_CONNECTION_KEEP_ALIVE,
```

Comment:
i would do the exact opposite on this condition
if that's "DATA / ERROR / COMPLETE" and req does not exists, then send cancel again.